src/main/java/com/targomo/client/api/pojo/RoutingOptions.java

```diff
@@ -51,10 +51,19 @@ public class RoutingOptions {
 
     @Override //overriding hash code since the has would change with each execution for enums (travelType)
     public int hashCode(){
-        return Objects.hash(maxEdgeWeight, travelType.getKey(), travelTimeFactors, reverse, rushHour, time, date, frame, arrivalOrDepartureDuration,
+        return Objects.hash(maxEdgeWeight, travelType.ordinal(), travelTimeFactors, reverse, rushHour, time, date, frame, arrivalOrDepartureDuration,
                 elevationEnabled, bikeSpeed, bikeDownhill, bikeUphill, walkSpeed, walkDownhill, walkUphill, allowPrivateAndServiceRoads,
                 trafficJunctionPenalty, trafficSignalPenalty, trafficLeftTurnPenalty, trafficRightTurnPenalty,
                 maxTransfers, maxWalkingTimeFromSource, maxWalkingTimeToTarget, avoidTransitRouteTypes,
                 maxSnapDistance, excludeEdgeClassesFromSnapping, useAreaSnapping, snapRadius, includeSnapDistance);
     }
+
+    @Override
+    public boolean equals(Object other){
+        if(!(other instanceof RoutingOptions)){
+            return false;
+        }
+        RoutingOptions that = (RoutingOptions) other;
+        return this.hashCode() == that.hashCode();
+    }
```

Comment:
this is dangerous. to have equals be based on the hash. why do we not want to use the lombok equals method?

Reply:
you created the `hashcode` function because of the traveltype enum. that's why I used it
why this is dangerous? what was dangerous was to use different function, which led to my problem of having two same objects with equals returning false.

Reply:
I edited it with an auto-generated function